en/lessons/basics/collections.md

```diff
@@ -3,120 +3,224 @@ version: 1.3.1
+
+Tuples are the first kind of collection you are introduced to in haskell. This is because they are simple, primitive, and have a terse, flexible, built in syntax. Tuples are sometimes referred to as anonymous records, but instead of referencing fields by name you reference them by position in the structure. Tuples can contain an arbitrary number of items, but we will focus on the 2-tuple since haskell has good built in support for them. However, if you are interested this is what an 8-tuple looks like
+
+```haskell
+Prelude> :t ('0','1','2','3','4','5','6',"8-tuple")
+('0','1','2','3','4','5','6',"8-tuple")
+  :: (Char, Char, Char, Char, Char, Char, Char, [Char])
+```
+__Note__: `:t` in the ghci repl shows the type of the value, it will print out a message of the form `term :: Type`
+
+### Constructing Tuples
+
+We can construct tuples using a familiar syntax `(a,b)`.
+
+```haskell
+Prelude> myTuple = (1 :: Int,"hello")
+Prelude> :t myTuple
+myTuple :: (Int, [Char])
+```
+
+### Consuming Tuples
+
+We can consume tuples by breaking them apart. With a 2-tuple the options are pretty simple. They have the predictable names `fst` and `snd`
+
+```haskell
+Prelude> fst myTuple
+1
+Prelude> snd myTuple
+"hello"
+```
+
+We can also pattern match on tuples to extract their contents.
+
+```haskell
+Prelude> (\(a,b) -> a + 2) $ myTuple
+3
+Prelude> (\(a,b) -> b ++ " world") $ myTuple
+"string world"
+```
+
+### Limitations of Tuples
+
+There are a couple of limitations with tuples in haskell. First, they are closed; you cannot add an additional item once the tuple of length n has been defined. Second, the types are fixed; once a tuple has a concrete type you cannot add an element of a different type. You can get around this with specialized functions, but they do not generalize to all tuples lengths.
+
+Here is an example of increasing the tuple's length.
+
+```haskell
+Prelude> twoTupleToThreeTuple c (a,b) = (a,b,c)
+Prelude> twoTupleToThreeTuple () myTuple
+(1,"world",())
+```
+
+Here is an example of changing a tuple's type.
+
+```haskell
+Prelude> :t myTuple
+myTuple :: (Int, [Char])
+Prelude> :t substituteFst "hello" myTuple
+substituteFst "hello" myTuple :: ([Char], [Char])
+Prelude> substituteFst "hello" myTuple
+("hello","world")
+```
 ## Lists
 
-Lists are simple collections of values which may include multiple types; lists may also include non-unique values:
+From a usability perspective, Lists solve the problem of extensibility that tuples face, but they can only contain one type (in other words they are homogenous). Lists also have a special built in syntax.
 
-```elixir
-iex> [3.14, :pie, "Apple"]
-[3.14, :pie, "Apple"]
+```haskell
+Prelude> [1,2,3,4]
+[1,2,3,4]
 ```
 
-Elixir implements list collections as linked lists.
+Haskell implements list collections as linked lists.
 This means that accessing the list length is an operation that will run in linear time (`O(n)`).
 For this reason, it is typically faster to prepend than to append:
```

Comment:
Maybe this section will work better after explaining that list is a recursive data type